Userland/Libraries/LibArchive/Tar.cpp

```diff
@@ -24,10 +24,12 @@ unsigned TarFileHeader::expected_checksum() const
     return checksum;
 }
 
-void TarFileHeader::calculate_checksum()
+ErrorOr<void> TarFileHeader::calculate_checksum()
 {
     memset(m_checksum, ' ', sizeof(m_checksum));
-    VERIFY(DeprecatedString::formatted("{:06o}", expected_checksum()).copy_characters_to_buffer(m_checksum, sizeof(m_checksum)));
+    bool copy_successful = TRY(String::formatted("{:06o}", expected_checksum())).bytes_as_string_view().copy_characters_to_buffer(m_checksum, sizeof(m_checksum));
+    VERIFY(copy_successful);
```

Comment:
Should this be a `return Error::from_string_literal("...");` instead? It seems strange to assert on failure for this.

Reply:
If our checksum calculation returns something bigger than what can be represented in the checksum field, that's probably not user error. The `VERIFY` there is fine in my opinion.

Reply:
Ah, ok. I didn't understand what that line was doing, but that makes sense. Thanks!